packages/react-charting/UnitTests/LineChartUT.test.tsx

```diff
@@ -0,0 +1,533 @@
+        {
+          x: 1,
+          y: 20,
+        },
+        {
+          x: 2,
+          y: 30,
+        },
+        {
+          x: 3,
+          y: 40,
+        },
+        {
+          x: 4,
+          y: 50,
+        },
+        {
+          x: 5,
+          y: 60,
+        },
+        {
+          x: 6,
+          y: 70,
+        },
+        {
+          x: 7,
+          y: 80,
+        },
+        {
+          x: 8,
+          y: 90,
+        },
+      ],
+      color: DefaultPalette.blue,
+      lineOptions: {
+        lineBorderWidth: '4',
+      },
+    },
+    {
+      legend: 'All',
+      data: [
+        {
+          x: 0,
+          y: 100,
+        },
+        {
+          x: 1,
+          y: 200,
+        },
+        {
+          x: 2,
+          y: 300,
+        },
+        {
+          x: 3,
+          y: 400,
+        },
+        {
+          x: 4,
+          y: 500,
+        },
+        {
+          x: 5,
+          y: 600,
+        },
+        {
+          x: 6,
+          y: 700,
+        },
+      ],
+      color: DefaultPalette.green,
+      lineOptions: {
+        lineBorderWidth: '4',
+      },
+    },
+    {
+      legend: 'single point',
+      data: [
+        {
+          x: 0,
+          y: 100,
+        },
+      ],
+      color: DefaultPalette.yellow,
+    },
+    {
+      legend: 'two points',
+      data: [
+        {
+          x: 0,
+          y: 100,
+        },
+        {
+          x: 1,
+          y: 200,
+        },
+      ],
+      color: DefaultPalette.green,
+    },
+    {
+      legend: 'three points',
+      data: [
+        {
+          x: 0,
+          y: 100,
+        },
+        {
+          x: 1,
+          y: 200,
+        },
+        {
+          x: 2,
+          y: 300,
+        },
+      ],
+      color: DefaultPalette.blue,
+    },
+    {
+      legend: 'four points',
+      data: [
+        {
+          x: 0,
+          y: 100,
+        },
+        {
+          x: 1,
+          y: 200,
+        },
+        {
+          x: 2,
+          y: 300,
+        },
+        {
+          x: 3,
+          y: 400,
+        },
+      ],
+      color: DefaultPalette.blue,
+    },
+  ],
+};
+
+runTest('Unit tests for _getPath function', () => {
+  testWithoutWait(
+    'Should return correct path when multiple shapes for points are not allowed and widthRatio and w equal to 1',
+    LineChart,
+    { data: basicChartPoints, allowMultipleShapesForPoints: false, strokeWidth: 0.5 },
+    container => {
+      const svg = container.querySelector('svg');
+      expect(svg).not.toBeNull();
+      const paths = svg!.querySelectorAll('path');
+      expect(paths).toHaveLength(6);
+      expect(paths[0]!.getAttribute('d')).toBe(`M-20.5 -17.065217391304344
+     A0.5 0.5 0 1 0 -19.5 -17.065217391304344
+     M-20.5 -17.065217391304344
+     A 0.5 0.5 0 1 1 -19.5 -17.065217391304344
+     `);
+      expect(paths[1]!.getAttribute('d')).toBe(`M15.5 12.826086956521738
+     A0.5 0.5 0 1 0 16.5 12.826086956521738
```

Comment:
I remember there was an intermediate issue in the path value. Did it get resolved. What was the issue

Reply:
yes, the issue came because of d3 library upgrade

Reply:
can you explain more which d3 library upgrade caused the issue